packages/react/__tests__/custom-cache.js

```diff
@@ -38,3 +45,46 @@ test('with custom plugins', () => {
     />
   `)
 })
+
+test('Global should "inherit" sheet class from the cache', () => {
+  // https://github.com/emotion-js/emotion/issues/2675
+  let cache = createCache({
+    key: 'test',
+    speedy: false
+  })
+  class MySheet extends StyleSheet {
```

Comment:
I've noticed that you were constructing different sheet classes in the constructor of your custom implementation but perhaps that's not really needed. You can easily differentiate hashed/main and global sheets by just checking the `options.key.endsWith('-global')`. Based on that you can just assign some private property and branch the logic in the `insert` method. This way you only have to write a single `StyleSheet` class which might simplify your implementation.

Reply:
oh, I need to get the un-global version of the sheet so that's the easiest way, any way, thanks!